src/fetcher.js

```diff
@@ -788,22 +768,29 @@ class Fetcher {
 
     return this._fetch(actualProxyURI, options)
       .then(response => this.handleResponse(response, docuri, options),
-            error =>
-              // handleError expects a response so we fake some important bits.
-              this.handleError({
+            error => {
+              let dummyResponse = {
                 url: actualProxyURI,
                 status: 999, // @@ what number/string should fetch failures report?
                 statusText: (error.name || 'network failure') + ': ' +
                   (error.errno || error.code || error.type),
                 responseText: error.message,
-                headers: Headers(),
+                headers: {},  // Headers() ???
                 ok: false,
                 body: null,
                 bodyUsed: false,
                 size: 0,
                 timeout: 0
-              }, docuri, options)
-      )
+              }
+              console.log('Fetcher: Non-HTTP fetch error ' + error)
+              return this.failFetch(options, 'fetch failed: ' + error, 999, dummyResponse) // Fake status code: fetch exception
```

Comment:
It might be useful to have a reference to the error when downstream wants to do some logic based on the exception type.